aiohasupervisor/models/backups.py

```diff
@@ -0,0 +1,169 @@
+"""Models for Supervisor backups."""
+
+from abc import ABC
+from dataclasses import dataclass
+from datetime import datetime
+from enum import StrEnum
+
+from .base import Request, ResponseData
+
+# --- ENUMS ----
+
+
+class BackupType(StrEnum):
+    """BackupType type."""
+
+    FULL = "full"
+    PARTIAL = "partial"
+
+
+class Folder(StrEnum):
+    """Folder type."""
+
+    SHARE = "share"
+    ADDONS = "addons/local"
+    SSL = "ssl"
+    MEDIA = "media"
+
+
+# --- OBJECTS ----
+
+
+@dataclass(frozen=True, slots=True)
+class BackupContent(ResponseData):
+    """BackupContent model."""
+
+    homeassistant: bool
+    addons: list[str]
+    folders: list[str]
+
+
+@dataclass(frozen=True)
+class BackupBaseFields(ABC):
+    """BackupBaseFields ABC type."""
+
+    slug: str
+    name: str
+    date: datetime
+    type: BackupType
+    size: float
+    location: str | None
+    protected: bool
+    compressed: bool
+
+
+@dataclass(frozen=True, slots=True)
+class Backup(BackupBaseFields, ResponseData):
+    """Backup model."""
+
+    content: BackupContent
+
+
+@dataclass(frozen=True, slots=True)
+class BackupAddon(ResponseData):
+    """BackupAddon model."""
+
+    slug: str
+    name: str
+    version: str
+    size: float
+
+
+@dataclass(frozen=True, slots=True)
+class BackupComplete(BackupBaseFields, ResponseData):
+    """BackupComplete model."""
+
+    supervisor_version: str
+    homeassistant: str
+    addons: list[BackupAddon]
+    repositories: list[str]
+    folders: list[str]
+    homeassistant_exclude_database: bool | None
+
+
+@dataclass(frozen=True, slots=True)
+class BackupList(ResponseData):
+    """BackupList model."""
+
+    backups: list[Backup]
+
+
+@dataclass(frozen=True, slots=True)
+class BackupsInfo(BackupList):
+    """BackupsInfo model."""
+
+    days_until_stale: int
+
+
+@dataclass(frozen=True, slots=True)
+class BackupsOptions(Request):
+    """BackupsOptions model."""
+
+    days_until_stale: int
+
+
+@dataclass(frozen=True, slots=True)
+class FreezeOptions(Request):
+    """FreezeOptions model."""
+
+    timeout: int
+
+
+@dataclass(frozen=True)
+class PartialBackupRestoreOptions(ABC):  # noqa: B024
+    """PartialBackupRestoreOptions ABC type."""
+
+    addons: set[str] | None = None
+    folders: set[Folder] | None = None
+    homeassistant: bool | None = None
+
+    def __post_init__(self) -> None:
+        """Validate at least one thing to backup/restore is included."""
+        if not any((self.addons, self.folders, self.homeassistant)):
+            raise TypeError(
```

Comment:
Hm so the only issue I have with this is I used `TypeError` here:
https://github.com/home-assistant-libs/python-supervisor-client/blob/08581a99e364879c22e931b1e21c8fc07bfe23f5/aiohasupervisor/models/base.py#L61

So switching to `ValueError` feels inconsistent. It also felt more like an issue with the instance as a whole rather then any particular value. Since omitting those fields is fine you just can't omit all of them.

Reply:
Hm ok I see `TypeError` is really for when you're given something of an incorrect type, not for a situation like this. I guess in that case I'll switch them both to `ValueError`.